### DISCUSSION TITLE
How to reconnect ws when waking up from sleep on smartphone?

### DISCUSSION THREAD

**Discussion by alyweb66** [enisdenjo/graphql-ws#580] [Q&A] [answered]
Hello,
I have a problem, when I go to my website, which is with React and Apollo server, from a smartphone, when it goes to sleep and I wait a little then I take it out of sleep and I arrive on the page of my site i have this error : `The connection to wss://<domain>/subscriptions has terminated unexpectedly. Some data may have been transferred.`

I tried lot of things i saw on the net like: 

```
const wsLink = new GraphQLWsLink(
  createClient({
    url: import.meta.env.VITE_SERVER_SUBSCRIPTION,
    lazy: true,
    keepAlive: 5000, 
    retryAttempts: 5, 
    on: {
      ping: (received) => {
        if (!received) {
 
          timedOut = setTimeout(() => {
            console.log('No response.');
            wsLink.client.terminate();
          }, 5000); 
        }
      },
      pong: (received) => {
        if (received) {
          console.log('Get response.');
          clearTimeout(timedOut); 
        }
      },
    },
  })
);
```
The temporary solution I found, is reload page:

```
const wsLink = new GraphQLWsLink(
	createClient({
		url: import.meta.env.VITE_SERVER_SUBSCRIPTION,
		retryAttempts: Infinity,
		shouldRetry: () => true,
		keepAlive: 10000,
		on: {
			error: () => {
				window.location.reload();
			},
		},
	})
);
```

And more but nothing worked for me...
Can you help me?
Thank you

**Comment by enisdenjo** [comment, marked as answer]
Hey there! If the client is not automatically retrying to connect - the socket is closing with a fatal problem. To change what's considered fatal in graphql-ws, you can use [the `shouldRetry` option](https://the-guild.dev/graphql/ws/docs/interfaces/client.ClientOptions#shouldretry). Hope that helps!

**Comment by alyweb66** [comment]
Thank you to your answer,

You mean i need to do something like that: 

```
shouldRetry: (errOrCloseEvent) => {
            if (errOrCloseEvent instanceof CloseEvent) {
               	// No retry for normal closure, going away, or abnormal closure
                const fatalCloseCodes = [1000, 1001, 1006]; // Normal Closure, Going Away, Abnormal Closure
                return !fatalCloseCodes.includes(errOrCloseEvent.code);
            }
            return true;
        },
```

**Reply by enisdenjo** [reply]
Yeah something like that. Also, code 1000 is "normal closure" and will never be retried so you can remove it from the list.

**Reply by alyweb66** [reply]
Ok i tried this, and that seem working, but that create a new problem with firefox, if i return false i have a subscription error when i refresh the page, i think it's because that close the websocket so that create an error, i think i must find a solution without return false...
